src/encoder.js

```diff
@@ -65,8 +65,15 @@ function encodeString(str, target, lut) {
       target.push(tags.STR_);
       encodeUInt(numBytes, target);
     }
-    // todo: find way around blowing callstack with huge strings
-    target.push.apply(target, utf8Bytes);
+    const APPLY_CHUNK_SIZE = 0xFFFF;
+    if (utf8Bytes.length > APPLY_CHUNK_SIZE) {
+      while (utf8Bytes.length > 0) {
+        [].push.apply(target, utf8Bytes.slice(0, APPLY_CHUNK_SIZE));
+        utf8Bytes = utf8Bytes.slice(APPLY_CHUNK_SIZE);
```

Comment:
Why the mutate-as-you-consume pattern instead of just incrementing a start pointer? Something like:
```javascript
for (var startIdx = 0; startIdx < utf8Bytes.length; startIdx += APPLY_CHUNK_SIZE) {
  [].push.apply(target, utf8Bytes.slice(startIdx, APPLY_CHUNK_SIZE));
}
```
Maybe it doesn't make a performance difference in practice, but it seems like it would in theory.